tests/CMakeLists.txt

```diff
@@ -34,7 +34,7 @@ endif()
 # please try to keep entries ordered =)
 add_library(gsl_tests_config INTERFACE)
 target_compile_options(gsl_tests_config INTERFACE
-    $<$<CXX_COMPILER_ID:MSVC>:
+    $<$<AND:$<PLATFORM_ID:Windows>,$<OR:$<CXX_COMPILER_ID:MSVC>,$<CXX_COMPILER_ID:Clang>>>:
```

Comment:
The logic got complicated here, looks like it's worth creating separate cases for msvc, clang-cl, clang, gcc options, and splitting the code accordingly, even if some options are repeated sometimes - it will help the code look intentional and maintainable.

For example, before, we had msvc vs the rest cases, now it looks like clang-cl will have a mix of msvc and clang options, so it looks like a logic error. Is it intentional?

Reply:
It was intentional. Clang-cl expects the flags from cl.exe, but anything not in MSVC, like the warnings (besides the sets `-Wall` and `-Wextra`) requires the 'normal' clang flags.

I'll make it a dual configuration, MSVC-like and GNUC-like.
(Like clang, Intel C++ also ships in two such platform-dependent versions.)
With a generator expression per compiler for specific flags. Some duplication, but it'll limit the if-statement nesting to two levels.

Reply:
Looks good!